pkg/util/protomarshal/protomarshal.go

```diff
@@ -210,7 +211,9 @@ func ApplyYAMLStrict(yml string, pb proto.Message) error {
 	return ApplyJSONStrict(string(js), pb)
 }
 
-func ShallowCopy(dst, src proto.Message) {
+// ShallowClone performs a shallow clone of the object. For a deep clone, use Clone.
+func ShallowClone[T proto.Message](src T) T {
+	dst := ptr.Empty[T]()
 	dm := dst.ProtoReflect()
 	sm := src.ProtoReflect()
 	if dm.Type() != sm.Type() {
```

Comment:
checking that the types match and performing a reset on dst seem unnecessary after the refactor